lens/database.ml

```diff
@@ -302,5 +336,78 @@ module Insert = struct
       (fmt_vals |> Format.pp_comma_list)
       v.values fmt_returning ()
 
+  let split v =
+    let { table; columns; values; returning } = v in
+    let f v =
+      let values = [ v ] in
+      { table; columns; values; returning }
+    in
+    List.map ~f values
+
+  let no_returning v = { v with returning = [] }
+
+  let exec_insert_returning_hack ~db data =
```

Comment:
There is also code accomplishing this in the main core/database.ml and driver modules.  It is unfortunate that our current architecture for splitting Links into core and lens libraries leads to repetition of code for essentially the same purpose.  It would be good to find a way to avoid this someday...

Reply:
There is code which does something similar, but it is a bit more primitive in the sense that it requires a links style insert query and does not always handle the multivalue case (and therefore not efficiently).